src/ray/util/util.cc

```diff
@@ -77,7 +82,8 @@ std::string EndpointToUrl(
     result = ss.str();
     break;
   }
-#ifdef BOOST_ASIO_HAS_LOCAL_SOCKETS
+
+#if BOOST_ASIO_HAS_LOCAL_SOCKETS
```

Comment:
I guessed a wrong value of this flag. It's always set to 1 on windows CI.

Due to the CI error message shows `sockaddr_un` is not defined, I believe we should add some including lines for it:
![image](https://user-images.githubusercontent.com/19473861/198649919-2bef2036-1e9b-41ec-b759-bb9c2307d97a.png)

I'm testing it, I believe it should be pass.